src/chfn.c

```diff
@@ -648,11 +648,20 @@ int main (int argc, char **argv)
 	 * name, or the name getlogin() returns.
 	 */
 	if (optind < argc) {
-		user = argv[optind];
+		if (is_valid_user_name (argv[optind])) {
+			user = xstrdup (argv[optind]);
+		}
+		else {
+			fprintf (stderr, _("%s: user '%s' does not exist\n"), Prog,
+			         user);
+			fail_exit (E_NOPERM);
+		}
+
```

Comment:
If you invert the conditional, the first condition would contain a `[[noreturn]]` call, which would make the `else` unnecessary.  Please do.

Reply:
This is fixed